File: src/me/krickl/memebotj/InternalCommands/CommandList.java
package me.krickl.memebotj.InternalCommands;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.UserHandler;

public class CommandList extends CommandHandler {

	public CommandList(String channel, String command, String dbprefix) {
		super(channel, command, dbprefix);
		this.setHelptext("Displays a command list");
	}

	@Override
	public void commandScript(UserHandler sender, ChannelHandler channelHandler, String[] data) {
		if(Memebot.useWeb()) {
			channelHandler.sendMessage("Commands: " + channelHandler.getChannelPageBaseURL()
						+ "/index.html || Use !help for further information || Documentation: https://github.com/unlink2/memebotj/blob/master/README.md",
						this.getChannelOrigin());
		} else {
			int index = 0;
			String output = "";
			try {
				index = Integer.parseInt(data[0]);
			} catch(ArrayIndexOutOfBoundsException e) {
				e.printStackTrace();
			} catch(NumberFormatException e) {
				e.printStackTrace();
			}

			for(int i = index * 10; i < channelHandler.getChannelCommands().size(); i++) {
				if(i > index * 10 + 10) {
					break;
				}
				output = output + " || " + channelHandler.getChannelCommands().get(i).getCommand();
			}

			channelHandler.sendMessage("Commands: " + output, this.getChannelOrigin());
		}
	}
}

File: src/me/krickl/memebotj/InternalCommands/UserCommands/CommandList.scala
package me.krickl.memebotj.InternalCommands.UserCommands

import me.krickl.memebotj.{ChannelHandler, CommandHandler, Memebot, UserHandler}

class CommandList(channel: String, command: String, dbprefix: String) extends CommandHandler(channel,
  command, dbprefix) {

  this.setHelptext(Memebot.formatText("COMMAND_LIST_SYNTAX", channelOriginHandler, null, this, true, Array()))

  override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
    try {
      if (Memebot.useWeb) {
        channelHandler.sendMessage(Memebot.formatText("COMMAND_LIST", channelHandler, sender, this, true, Array(channelHandler.getChannelPageBaseURL + "/index.html")), this.getChannelOrigin)
      } else {
        var index = 0
        var output = ""
        try {
          index = java.lang.Integer.parseInt(data(0))
        } catch {
          case e: ArrayIndexOutOfBoundsException => e.printStackTrace()
          case e: NumberFormatException => e.printStackTrace()
        }
        for (i <- index * 10 until channelHandler.getChannelCommands.size) {
          if (i > index * 10 + 10) {
            //break
          }
          output = output + " || " +
            channelHandler.getChannelCommands.get(i).getCommand
        }
        channelHandler.sendMessage("Commands: " + output, this.getChannelOrigin)
      }
    } catch {
      case e: NumberFormatException =>
        channelHandler.sendMessage(Memebot.formatText("COMMAND_LIST_ERROR", channelHandler, sender, this, true, Array()), this.getChannelOrigin)
        e.printStackTrace()
    }
  }
}
